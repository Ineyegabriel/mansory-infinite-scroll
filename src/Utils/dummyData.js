const DUMMY_DATA = {
  total: 1359084,
  totalHits: 500,
  hits: [
    {
      id: 5887065,
      pageURL:
        "https://pixabay.com/photos/rooster-chicken-head-cockscomb-5887065/",
      type: "photo",
      tags: "rooster, chicken, head",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/10/16/rooster-5887065_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd444a54a914f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd444a54a914f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6240,
      imageHeight: 4160,
      imageSize: 6153308,
      views: 4544,
      downloads: 4259,
      favorites: 17,
      likes: 56,
      comments: 54,
      user_id: 10327513,
      user: "NickyPe",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/11/30/11-19-23-84_250x250.jpg",
    },
    {
      id: 5887061,
      pageURL:
        "https://pixabay.com/photos/jerusalem-artichoke-sunflower-flower-5887061/",
      type: "photo",
      tags: "jerusalem artichoke, sunflower, flower",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/10/12/jerusalem-artichoke-5887061_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd444a54ad14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd444a54ad14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6240,
      imageHeight: 4160,
      imageSize: 3819636,
      views: 4063,
      downloads: 3885,
      favorites: 14,
      likes: 49,
      comments: 46,
      user_id: 10327513,
      user: "NickyPe",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/11/30/11-19-23-84_250x250.jpg",
    },
    {
      id: 5880859,
      pageURL:
        "https://pixabay.com/illustrations/hot-air-balloons-clouds-sky-heart-5880859/",
      type: "illustration",
      tags: "hot air balloons, clouds, sky",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/02/08/16/hot-air-balloons-5880859_150.png",
      previewWidth: 150,
      previewHeight: 107,
      webformatURL:
        "https://pixabay.com/get/53e8dd434257a514f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.png",
      webformatWidth: 640,
      webformatHeight: 457,
      largeImageURL:
        "https://pixabay.com/get/53e8dd434257a514f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.png",
      imageWidth: 4209,
      imageHeight: 3007,
      imageSize: 1248853,
      views: 3440,
      downloads: 2812,
      favorites: 41,
      likes: 67,
      comments: 39,
      user_id: 13452116,
      user: "Syaibatulhamdi",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/08/19/09-44-13-930_250x250.png",
    },
    {
      id: 5885023,
      pageURL:
        "https://pixabay.com/photos/butterfly-admiral-butterfly-wings-5885023/",
      type: "photo",
      tags: "butterfly, admiral, butterfly wings",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/03/16/55/butterfly-5885023_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd464a50af14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd464a50af14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 4000,
      imageSize: 4360278,
      views: 2963,
      downloads: 2865,
      favorites: 13,
      likes: 40,
      comments: 37,
      user_id: 1767157,
      user: "Capri23auto",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/12/27/05-34-04-344_250x250.jpg",
    },
    {
      id: 5880610,
      pageURL:
        "https://pixabay.com/illustrations/clouds-sun-sky-paper-sunlight-day-5880610/",
      type: "illustration",
      tags: "clouds, sun, sky",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/02/04/11/clouds-5880610_150.png",
      previewWidth: 150,
      previewHeight: 104,
      webformatURL:
        "https://pixabay.com/get/53e8dd434c53ac14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.png",
      webformatWidth: 640,
      webformatHeight: 442,
      largeImageURL:
        "https://pixabay.com/get/53e8dd434c53ac14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.png",
      imageWidth: 4209,
      imageHeight: 2907,
      imageSize: 1419415,
      views: 2793,
      downloads: 2335,
      favorites: 39,
      likes: 64,
      comments: 32,
      user_id: 13452116,
      user: "Syaibatulhamdi",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/08/19/09-44-13-930_250x250.png",
    },
    {
      id: 5886726,
      pageURL:
        "https://pixabay.com/photos/lily-flower-blossom-bloom-pistils-5886726/",
      type: "photo",
      tags: "lily, flower, blossom",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/07/38/lily-5886726_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd454d50aa14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd454d50aa14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 4000,
      imageSize: 2291684,
      views: 1976,
      downloads: 1896,
      favorites: 5,
      likes: 35,
      comments: 34,
      user_id: 12752456,
      user: "Mammiya",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/11/17/14-02-01-560_250x250.jpg",
    },
    {
      id: 5887412,
      pageURL:
        "https://pixabay.com/photos/flower-cutleaf-coneflower-blossom-5887412/",
      type: "photo",
      tags: "flower, cutleaf coneflower, blossom",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/12/40/flower-5887412_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd444e53ae14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd444e53ae14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 4000,
      imageSize: 3488058,
      views: 556,
      downloads: 509,
      favorites: 10,
      likes: 30,
      comments: 26,
      user_id: 1767157,
      user: "Capri23auto",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/12/27/05-34-04-344_250x250.jpg",
    },
    {
      id: 5883456,
      pageURL:
        "https://pixabay.com/photos/house-mountains-moon-alps-fields-5883456/",
      type: "photo",
      tags: "house, mountains, moon",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/03/02/09/house-5883456_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd404e57aa14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd404e57aa14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 3000,
      imageHeight: 2000,
      imageSize: 1084169,
      views: 9551,
      downloads: 9131,
      favorites: 15,
      likes: 54,
      comments: 59,
      user_id: 7009216,
      user: "susan-lu4esm",
      userImageURL:
        "https://cdn.pixabay.com/user/2021/01/05/01-32-41-770_250x250.jpg",
    },
    {
      id: 5888244,
      pageURL:
        "https://pixabay.com/photos/tulip-shadow-wall-flower-bud-5888244/",
      type: "photo",
      tags: "tulip, shadow, wall",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/17/28/tulip-5888244_150.jpg",
      previewWidth: 100,
      previewHeight: 150,
      webformatURL:
        "https://pixabay.com/get/53e8dd4b4856a814f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 427,
      webformatHeight: 640,
      largeImageURL:
        "https://pixabay.com/get/53e8dd4b4856a814f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 3447,
      imageHeight: 5163,
      imageSize: 1770882,
      views: 2458,
      downloads: 2385,
      favorites: 7,
      likes: 40,
      comments: 35,
      user_id: 1130890,
      user: "Peggychoucair",
      userImageURL:
        "https://cdn.pixabay.com/user/2018/12/28/22-17-08-435_250x250.jpg",
    },
    {
      id: 5888959,
      pageURL:
        "https://pixabay.com/photos/owl-snowy-owl-bird-white-owl-5888959/",
      type: "photo",
      tags: "owl, snowy owl, bird",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/21/46/owl-5888959_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd4b4357a514f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd4b4357a514f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 4000,
      imageHeight: 2667,
      imageSize: 1636825,
      views: 2382,
      downloads: 2291,
      favorites: 6,
      likes: 33,
      comments: 33,
      user_id: 6508325,
      user: "birgl",
      userImageURL:
        "https://cdn.pixabay.com/user/2021/01/08/07-27-16-186_250x250.jpg",
    },
    {
      id: 5878360,
      pageURL:
        "https://pixabay.com/photos/cosmea-cosmos-composites-blossom-5878360/",
      type: "photo",
      tags: "cosmea, cosmos, composites",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/01/11/55/cosmea-5878360_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8d24b4954ac14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8d24b4954ac14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 4000,
      imageSize: 4019734,
      views: 2120,
      downloads: 2064,
      favorites: 9,
      likes: 29,
      comments: 25,
      user_id: 1767157,
      user: "Capri23auto",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/12/27/05-34-04-344_250x250.jpg",
    },
    {
      id: 5886122,
      pageURL: "https://pixabay.com/photos/couple-rocks-fantasy-moon-5886122/",
      type: "photo",
      tags: "couple, rocks, fantasy",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/02/06/couple-5886122_150.jpg",
      previewWidth: 137,
      previewHeight: 150,
      webformatURL:
        "https://pixabay.com/get/53e8dd454b50ae14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 585,
      webformatHeight: 640,
      largeImageURL:
        "https://pixabay.com/get/53e8dd454b50ae14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 3198,
      imageHeight: 3500,
      imageSize: 780179,
      views: 2095,
      downloads: 1993,
      favorites: 10,
      likes: 35,
      comments: 39,
      user_id: 7009216,
      user: "susan-lu4esm",
      userImageURL:
        "https://cdn.pixabay.com/user/2021/01/05/01-32-41-770_250x250.jpg",
    },
    {
      id: 5862204,
      pageURL:
        "https://pixabay.com/illustrations/new-year-greetings-2021-clock-5862204/",
      type: "illustration",
      tags: "new year, greetings, 2021",
      previewURL:
        "https://cdn.pixabay.com/photo/2020/12/26/19/29/new-year-5862204_150.jpg",
      previewWidth: 150,
      previewHeight: 85,
      webformatURL:
        "https://pixabay.com/get/53e8d3414852a814f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 362,
      largeImageURL:
        "https://pixabay.com/get/53e8d3414852a814f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 3390,
      imageSize: 1321557,
      views: 126832,
      downloads: 47294,
      favorites: 70,
      likes: 186,
      comments: 56,
      user_id: 7438739,
      user: "jeffjacobs1990",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/11/13/04-26-15-936_250x250.jpg",
    },
    {
      id: 5890821,
      pageURL:
        "https://pixabay.com/photos/sky-clouds-cirrocumulus-cloudy-5890821/",
      type: "photo",
      tags: "sky, clouds, cirrocumulus",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/05/11/01/sky-5890821_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dc434250ad14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dc434250ad14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 5568,
      imageHeight: 3712,
      imageSize: 4049490,
      views: 502,
      downloads: 457,
      favorites: 9,
      likes: 31,
      comments: 26,
      user_id: 3764790,
      user: "enriquelopezgarre",
      userImageURL:
        "https://cdn.pixabay.com/user/2021/01/06/11-11-21-272_250x250.jpg",
    },
    {
      id: 5870169,
      pageURL: "https://pixabay.com/photos/marguerite-flower-dewdrops-5870169/",
      type: "photo",
      tags: "marguerite, flower, dewdrops",
      previewURL:
        "https://cdn.pixabay.com/photo/2020/12/29/10/39/marguerite-5870169_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8d2434b54a514f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8d2434b54a514f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 4000,
      imageSize: 1966249,
      views: 6553,
      downloads: 6340,
      favorites: 9,
      likes: 48,
      comments: 48,
      user_id: 12752456,
      user: "Mammiya",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/11/17/14-02-01-560_250x250.jpg",
    },
    {
      id: 5097557,
      pageURL:
        "https://pixabay.com/photos/green-meadow-nuns-light-dawn-grass-5097557/",
      type: "photo",
      tags: "green meadow, nuns, light",
      previewURL:
        "https://cdn.pixabay.com/photo/2020/04/26/23/46/green-meadow-5097557_150.jpg",
      previewWidth: 150,
      previewHeight: 94,
      webformatURL:
        "https://pixabay.com/get/53e0dc444f57ab14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 402,
      largeImageURL:
        "https://pixabay.com/get/53e0dc444f57ab14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 5383,
      imageHeight: 3383,
      imageSize: 3558131,
      views: 6307,
      downloads: 6131,
      favorites: 7,
      likes: 34,
      comments: 37,
      user_id: 6355831,
      user: "pasja1000",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/12/08/21-33-27-942_250x250.jpg",
    },
    {
      id: 5870088,
      pageURL:
        "https://pixabay.com/photos/coast-sea-fog-rocky-rocky-coast-5870088/",
      type: "photo",
      tags: "coast, sea, fog",
      previewURL:
        "https://cdn.pixabay.com/photo/2020/12/29/10/07/coast-5870088_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8d2434a5aa414f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8d2434a5aa414f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6016,
      imageHeight: 4016,
      imageSize: 4078114,
      views: 24713,
      downloads: 17650,
      favorites: 52,
      likes: 81,
      comments: 0,
      user_id: 15593888,
      user: "BernhardJaeck",
      userImageURL:
        "https://cdn.pixabay.com/user/2020/12/29/09-30-04-754_250x250.jpg",
    },
    {
      id: 5888958,
      pageURL:
        "https://pixabay.com/photos/nutria-animals-water-coypu-rodent-5888958/",
      type: "photo",
      tags: "nutria, animals, water",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/21/46/nutria-5888958_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8dd4b4357a414f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8dd4b4357a414f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 4000,
      imageHeight: 2667,
      imageSize: 3038523,
      views: 1504,
      downloads: 1436,
      favorites: 4,
      likes: 30,
      comments: 32,
      user_id: 6508325,
      user: "birgl",
      userImageURL:
        "https://cdn.pixabay.com/user/2021/01/08/07-27-16-186_250x250.jpg",
    },
    {
      id: 5873636,
      pageURL:
        "https://pixabay.com/photos/impala-oxpecker-bird-antelope-5873636/",
      type: "photo",
      tags: "impala, oxpecker, bird",
      previewURL:
        "https://cdn.pixabay.com/photo/2020/12/30/14/26/impala-5873636_150.jpg",
      previewWidth: 150,
      previewHeight: 100,
      webformatURL:
        "https://pixabay.com/get/53e8d2404c51aa14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 427,
      largeImageURL:
        "https://pixabay.com/get/53e8d2404c51aa14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 4862,
      imageHeight: 3241,
      imageSize: 4857368,
      views: 23478,
      downloads: 20197,
      favorites: 36,
      likes: 69,
      comments: 22,
      user_id: 19712633,
      user: "matrishva",
      userImageURL: "",
    },
    {
      id: 5888961,
      pageURL:
        "https://pixabay.com/photos/white-flower-blossom-white-petals-5888961/",
      type: "photo",
      tags: "white, flower, blossom",
      previewURL:
        "https://cdn.pixabay.com/photo/2021/01/04/21/46/white-5888961_150.jpg",
      previewWidth: 150,
      previewHeight: 89,
      webformatURL:
        "https://pixabay.com/get/53e8dd4b4354ad14f1dc84609629327d123cdbec5a4c704f742f78d49145cc5b_640.jpg",
      webformatWidth: 640,
      webformatHeight: 381,
      largeImageURL:
        "https://pixabay.com/get/53e8dd4b4354ad14f6da8c7dda79367b1638dde05a586c48732e7bd7974bcd51bb_1280.jpg",
      imageWidth: 6000,
      imageHeight: 3568,
      imageSize: 1616557,
      views: 1242,
      downloads: 1183,
      favorites: 7,
      likes: 34,
      comments: 33,
      user_id: 6508325,
      user: "birgl",
      userImageURL:
        "https://cdn.pixabay.com/user/2021/01/08/07-27-16-186_250x250.jpg",
    },
  ],
};

export default DUMMY_DATA;