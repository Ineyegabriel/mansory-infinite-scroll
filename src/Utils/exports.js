export const masonryOptions = {  
    fitWidth: true,  
    columnWidth: 305,  
    gutter: 5  
};