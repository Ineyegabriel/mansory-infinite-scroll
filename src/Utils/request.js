const axios = require("axios");  
const APIKEY = "15373599-b05b8127e7dd2f4418ce544d3"

export const useGetImages = (page = 1) => {
    return axios.get(`https://pixabay.com/api/?page=${page}&key=${APIKEY}`)
      .then(({data, status}) =>{
        const {total,hits} = data;
        return {total, hits, status}
      })
}

    