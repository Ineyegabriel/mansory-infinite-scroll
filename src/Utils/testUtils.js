import {mount} from 'enzyme';
import { act } from '@testing-library/react';
/**
 * Function to implement the the find util from jest
 *  @param {string} wrapper
 * @param {string} val
 * @returns {ShallowWrapper}
 */
export const findByAtrr = (wrapper, val) => wrapper.find(`[data-test="${val}"]`);

/**
 * reusable wrapper that takes in two props, 
 * Name of the Hooks and any external prop you want to pass
 * And it mount the hook which can then be used for testing purposes
 * @param {function} hook 
 * @param {object} externalProps 
 * @returns {null}
 */
export const renderHook = (hook, externalProps={}) => {
    let results;
    const HookeWrapper = () => {
      results = hook();
      return null;
    };
    act(() => mount(<HookeWrapper/>));
    return results;
  };