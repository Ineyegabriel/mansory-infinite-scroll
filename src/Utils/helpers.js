import React, { useCallback, useState } from "react";
import { useGetImages } from "./request";

export const useGetImageHelper = () => {
  const [images, setImages] = useState([]);
  const [page, setPage] = useState(1);
  const [total, setTotal] = useState(0);
  const [initialized, setInitialized] = useState(false);
  const [error, setError] = useState(null);
  const [status, setStatus] = useState(0);
  
  const getAllImages = useCallback(
    (pageNumber = 1) => {
      const FetchingImages = async (done) => {
        try {
          const response = await useGetImages(page); 
          const {total,hits,status} = response;
          setStatus(status);
          let imgs = images.concat(hits);
          setImages(imgs);
          setTotal(total);
          pageNumber++;
          setPage(pageNumber);
          done()
        } 
        catch (error) {
          setError(error);
        }
      }
      FetchingImages();
    },
    [images, page]
  );

  React.useEffect(() => {
    if (!initialized){
      getAllImages();
      setInitialized(true);
    } 
  }, [initialized, getAllImages]);

  return {
    images,
    setImages,
    total,
    error,
    status,
    getAllImages
  };
};
