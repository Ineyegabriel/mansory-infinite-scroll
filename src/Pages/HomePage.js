import React from "react";
import { Link } from "react-router-dom";
import InfiniteScroll from "react-infinite-scroller";
import Masonry from "react-masonry-component";
import { masonryOptions } from "../Utils/exports";
import { useGetImageHelper } from "../Utils/helpers";
import Spinner from "react-bootstrap/Spinner";

const HomePage = () => {
  const { images, total, getAllImages } = useGetImageHelper();

  const template =
    images.length > 0 ? (
      images.map(({ id, previewURL,largeImageURL,user,views,imageWidth,imageHeight,downloads,tags }, index) => {
        return (
          <Link
            to={{
              pathname: "/preview",
              state: {
                id, previewURL,largeImageURL,user,views,imageWidth,imageHeight,downloads,tags,
              }
            }}
          >
            <div key={index} style={{ marginBottom: 10 }}>
              <img src={previewURL} style={{ width: 300 }} alt="" />
            </div>
          </Link>
        );
      })
    ) : (
      <Spinner animation="border" role="status">
        <span className="sr-only">Loading...</span>
      </Spinner>
    );
  return (
    <div className="page" data-test="component-wrapper">
      <h1 className="text-center">Home</h1>
      <InfiniteScroll
        pageStart={1}
        loadMore={getAllImages}
        hasMore={total > images.length}
      >
        <Masonry
          className={"grid"}
          elementType={"div"}
          options={masonryOptions}
          disableImagesLoaded={false}
          updateOnEachImageLoad={false}
          data-test="component-Infinite"
          style={{margin:"0 auto"}}
        >
          {template}
        </Masonry>
      </InfiniteScroll>
    </div>
  );
};
export default HomePage;
