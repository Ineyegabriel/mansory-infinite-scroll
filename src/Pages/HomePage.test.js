import React from "react";
import { mount } from "enzyme";
import HomePage from "./HomePage";
import { findByAtrr } from "../Utils/testUtils";
  
let wrapper;
beforeEach(() => {
  wrapper = mount(<HomePage/>);
})


describe("HomePage renders", () => {
 
  test("HomePage component renders", () => {
    let component = findByAtrr(wrapper, "component-wrapper");
    expect(component.length).toBe(1);
  });
});
describe('Infinite Scroll Component', () => {
    test("InfiniteScroll component renders", () => {
      let component = findByAtrr(wrapper, "component-Infinite");
        expect(component.length).not.toBeNull();
    });
      test("InfiniteScroll component to Match Snapshot", () => {
          let component = findByAtrr(wrapper, "component-Infinite");
          expect(component).toMatchSnapshot();  
      });
  });
