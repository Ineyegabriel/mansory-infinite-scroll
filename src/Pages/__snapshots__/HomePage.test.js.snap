// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infinite Scroll Component InfiniteScroll component to Match Snapshot 1`] = `ReactWrapper {}`;
